spectral/src/functions/infoLicenseApache2.ts

```diff
@@ -1,28 +1,39 @@
-import { IFunctionPaths, IFunctionResult } from "@stoplight/spectral";
+import { IFunctionResult } from "@stoplight/spectral-core";
+import type { JsonPath } from '@stoplight/types';
 
-export default (targetVal: any, _: any, paths: IFunctionPaths): IFunctionResult[] => {
-	const path = paths.target;
+
+export default (targetVal: any, _: any, context: any): IFunctionResult[] => {
+	// console.log("targetVal: ")
+	// console.log(targetVal)
+	// console.log("\ncontext.path: ") 
+	// console.log(context.path)
+	let p: JsonPath = ['info']
+	// console.log("p: ")
+	// console.log(p)
 	if (!targetVal) {
 		return [
 			{
-				message: "OpenAPI `license` object is required",
-				path: path,
+				message: "OpenAPI `license` object is required\n in path: ",
+				// path: [...p, 'info'],
+				path: p
 			}
 		]
 	}
+
 	const results: IFunctionResult[] = [];
 	const expectName = 'Apache 2.0'
```

Comment:
There are many classes of validation on openapi schema.
Some validation can be done using ajv (https://www.npmjs.com/package/ajv)/jsonp validation without including javascript. Spectral embeds ajv for that reason. This validator is good to have however utilizing more generic validator can be good example for future `static` rules.

Reply:
Example:

https://meta.stoplight.io/docs/spectral/e5b9616d6d50c-custom-rulesets

```
rules:
  my-rule-name:
    description: Tags must have a description.
    given: $.info[*]
    severity: error
    then:
      field: license
      function: pattern
      functionOptions:
        match: "^Apache 2.0$"

Reply:
Obviously this is going to return quite cryptic error so your code is still much better in this regard.